src/Simulation/Case/SimulationCase.cpp

```diff
@@ -32,4 +32,7 @@ SimulationCase::SimulationCase(std::string ini_base, const MCSimExecutor& mc_sim
   // Global Environment
   glo_env_ = new GlobalEnvironment(&sim_config_);
 }
-SimulationCase::~SimulationCase() { delete glo_env_; }
+SimulationCase::~SimulationCase() {
+  delete glo_env_;
+  delete sim_config_.main_logger_;
```

Comment:
せっかく`sim_config_`はポインタじゃないので，やるとしても`SimulationConfig`のデストラクタでやりたいような気がするんですがどうでしょう？
それとも，`SimulationConfig::main_logger_`は結構他でも雑に弄られててそうするとdouble freeしちゃう，とかだったりしますかね．
https://github.com/ut-issl/s2e-core/blob/d1aa12ed2444b3fe826c3f8813cae6cc5f1adc60/src/Simulation/Case/SimulationCase.h#L27-L28

Reply:
Currently, the `SimulationConfig` is a defined as struct not as class. So it doesn't have destructor.  
Maybe we need large modification for this in future...

Reply:
Hmm...I think the only difference between `struct` and `class` is default access rights(`public` or `private`).
So, it can technically have destructor(but I agree with we will need large modification).

Reply:
Thank you. I didn't know that the struct also has destructor.  
I defined the `delete` in the destructor of `SimulationCase`.